### ISSUE TITLE
The compiler gets errors when I try to modify a method

### ISSUE THREAD

**Issue by orion2087** [0xd4d/dnSpy#796]
First of all, I want to point out that I am a beginner. I am trying to modify a C# class with dnspy but the compiler gets errors. I don't know if I am doing it correctly. The RESULT is empty = "".
I want to modify a method in order that it will not get a non-empty string and gets any string. What I should do? This is the method:

private string ReadFile(bool returnEnscryptedString)
		{
			string result;
			try
			{
				result = ((frmMain.g)<Module>.Invoke(1225))(this, returnEnscryptedString);
			}
			catch
			{
				throw;
			}
			return result;
		}

**Comment by 0xd4d** [comment]
Use the IL editor or use a deobfuscator first.

**Comment by orion2087** [comment]
It was because I am not practiced of IL editor, and for this reason I needed to compile it modifying the code via C# or VB. However, I resolved the problem with a lot of effort using an HexEditor, since when I tried to save the module, the software did not work. The module results of a lower size than the original. I do not know if dnSpy reduce the file or it's because many lines of codes are not produced in the final assembly.
